src/ray/core_worker/task_event_buffer.cc

```diff
@@ -177,45 +197,143 @@ void TaskEventBufferImpl::Stop() {
 
 bool TaskEventBufferImpl::Enabled() const { return enabled_; }
 
-void TaskEventBufferImpl::AddTaskEvent(std::unique_ptr<TaskEvent> task_event) {
+void TaskEventBufferImpl::AddTaskProfileEvent(TaskProfileEvent e) {
```

Comment:
adding task events to thread local buffer.

Reply:
Rather than allocating task events with `std::unique_ptr`, we use std::move into the local buffer to achieve better caching and memory perf, instead of allocating much small memory with `new`. 
As a side effect, this would require us to have functions for different derived classes of `TaskEvent`.

Reply:
Actually - let me validate this since now I have a baseline that works ^ 
I am not entirely sure what's the cost of reverting back to using pointers actually, never tested this in isolation.